src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Android.cs

```diff
@@ -529,5 +532,78 @@ public bool OnEditorAction(TextView v, [GeneratedEnum] ImeAction actionId, KeyEv
 			// Otherwise we return true to allow the focus to change correctly.
 			return actionId != ImeAction.ImeNull;
 		}
+
+		partial void OnTextCharacterCasingChangedPartial(DependencyPropertyChangedEventArgs e)
+		{
+			if (_textBoxView == null)
+			{
+				return;
+			}
+
+			var casing = (CharacterCasing)e.NewValue;
+
+			UpdateCasing(casing);
+		}
+
+		private void UpdateCasing(CharacterCasing characterCasing)
+		{
+			var currentFilters = _textBoxView.GetFilters()?.ToList() ?? new List<IInputFilter>();
+
+			//Remove any casing filters before applying new ones.
+			currentFilters.Remove(a => a is InputFilterAllLower || a is InputFilterAllCaps);
+
+			switch (characterCasing)
+			{
+				case CharacterCasing.Lower:
+					var lowerFilter = new List<IInputFilter>(currentFilters)
+										{
+											new InputFilterAllLower()
+										};
+					_textBoxView.SetFilters(lowerFilter.ToArray());
+
+					break;
+				case CharacterCasing.Upper:
+					var upperFilter = new List<IInputFilter>(currentFilters)
+										{
+											new InputFilterAllCaps()
+										};
+					_textBoxView.SetFilters(upperFilter.ToArray());
+
+					break;
+				case CharacterCasing.Normal:
+					break;
+				default:
+					break;
+			}
+		}
+	}
+
+	class InputFilterAllLower : InputFilterAllCaps
+	{
+		public override Java.Lang.ICharSequence FilterFormatted(Java.Lang.ICharSequence source, int start, int end, ISpanned dest, int dstart, int dend)
+		{
+			for (var x = start; x < end; x++)
+			{
+				if (char.IsUpper(source.ElementAt(x)))
+				{
+					var v = new char[end - start];
+					TextUtils.GetChars(source.ToString(), start, end, v, 0);
+					var s = new string(v).ToLower();
```

Comment:
Codacy found an issue: [Use the overload that takes a 'CultureInfo' or 'IFormatProvider' parameter.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6171516)